drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -524,312 +500,327 @@
-      header when the IOAM Option-Type is an Integrity Protected Option-Type.
-      It is defined as follows:</t>
+      <t>The Direct Export (DEX) Option-Type <xref target="RFC9326"/> is not
+      covered by the Integrity Protection Method defined in this document (see
+      <xref target="IPM"/>). Indeed, this document focuses on the integrity
+      protection of IOAM-Data-Fields, while DEX does not have IOAM-Data-Fields
+      by definition. Therefore, DEX is considered out of the scope of this
+      document.</t>
 
-      <t><figure title="IOAM Integrity Protection Header">
-            <artwork>
+      <t>The IOAM Integrity Protection header follows the IOAM Option-Type
+      header and precedes IOAM-Data-Fields, when the IOAM Option-Type is an
+      Integrity Protected Option-Type. It is defined as follows:</t>
 
+      <t><figure title="IOAM Integrity Protection header">
+        <artwork>
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|Signature-suite|  Nonce length |           Reserved            |
+|   Method-ID   |  Nonce Length |           Reserved            |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |                                                               |
 ~                             Nonce                             ~
 |                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |                                                               |
-~                           Signature                           ~
+~                  Integrity Check Value (ICV)                  ~
 |                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-
-
-</artwork>
-          </figure></t>
+        </artwork>
+      </figure></t>
 
       <t><list style="hanging">
-        <t hangText="Signature-suite:">8-bit unsigned integer.
-        This field defines the algorithm pair used to compute the digest
-        and the signature over the IOAM-Data-Fields, as defined in <xref
-        target="IOAM-IP-registry"/>.</t>
-
-        <t hangText="Nonce length:">8-bit unsigned integer. This field
-        specifies the length of the Nonce in octets.</t>
-
-        <t hangText="Reserved:">16-bit Reserved field. MUST be set to zero
+        <t hangText="Method-ID:">8-bit unsigned integer.
+        It defines the Integrity Protection Method to compute the Integrity
+        Check Value (ICV) field. If a node encounters an unknown value, it MUST
+        NOT change the contents of the IOAM Integrity Protection header and MUST
+        NOT change the contents of the IOAM-Data-Fields. In other words, the
+        node MUST NOT process the IOAM Option-Type.
+        See <xref target="IOAM-IP-registry"/>.</t>
```

Comment:
Indeed. I think it's expected. If a node does not implement an Integrity Protection Method (identified by the "Method-ID" field), then it is not able to update the ICV. Should it add its IOAM-Data-Fields no matter what, the Validator would consider the whole thing as altered. So I think it's probably better to force a node not to add its data in that case. Should I add a sentence to reflect your remark? Thoughts on the above?

Reply:
Hmmm... Assume you have a node that cannot do ICV but it does support IOAM. It could update the IOAM-data.
Then it is up to the validator to decide whether the info would still be used or not.
IMHO we don't need to be overly restrictive.

Reply:
What I fear is to have either (i) interop issues or (ii) an easy kill for compromised nodes. The integrity validation will never work in both cases (e.g., a new method is implemented by some of your nodes, not all; or malicious behaviors on compromised nodes). Of course the Validator can do what it wants with IOAM-Data-Fields, but what's the most probable outcome for an "invalid integrity" signal...? Which would be equivalent to "no IOAM at all for the node" IMHO. But I definitely understand why your opinion makes sense too. I guess it all comes down to whether we still want IOAM to "work" as expected, regardless the integrity validation is broken.

Reply:
I suggest to keep it as is for -09 and ask feedback from the WG on whether they think A or B is more appropriate. WDYT?

Reply:
Works for me.